modules/runner/src/integrationTest/java/org/apache/ignite/internal/runner/app/ItIgniteNodeRestartTest.java

```diff
@@ -878,6 +879,48 @@ public void nodeWithDataTest() throws InterruptedException {
         checkTableWithData(ignite, TABLE_NAME);
     }
 
+    /**
+     * Restarts the node which stores some data.
+     */
+    @Test
+    public void nodeWithDataAndIndexRebuildTest() throws InterruptedException {
+        IgniteImpl ignite = startNode(0);
+
+        createTableWithData(List.of(ignite), TABLE_NAME, 1, 20);
+
+        TableImpl table = (TableImpl) ignite.tables().table(TABLE_NAME);
+
+        InternalTableImpl internalTable = (InternalTableImpl) table.internalTable();
+
+        for (int i = 0; i < 20; i++) {
+            // Flush data on disk, so that we will have a snapshot to read on restart.
+            internalTable.storage().getMvPartition(i).flush().join();
```

Comment:
Let's not use `join` here and use an assertion instead

Reply:
Also, these flushes can be done in parallel, no need to wait for them sequentially